### DISCUSSION TITLE
Splitting the Spec and the Primer by conformance class

### DISCUSSION THREAD

**Discussion by elf-pavlik** [solid/data-interoperability-panel#134] [General]
While working on TS implementation, separating aspect related to Applications and Authorization Agent is proving vert helpful. I would like to propose that we consider splitting the specification into.

* Data Model
* Applications
* Authorization Agent

As well as the primer into:

* Applications
* Authorization Agent

Data Model would be common base which both Applications and Authorization Agent would rely on. Audience specific specs would include all the requirements specific to that conformance class.

**Comment by ericprud** [comment]
[unresearched gut reaction] I can imagine that separating them would make the narrative more difficult to author or follow. Would typographical conventions (e.g. colors) do the job?

**Reply by elf-pavlik** [reply]
I think that for developers who just want to implement library in certain language to write applications. There's just to much information which they don't really need. We could start by marking sections with which conformance class they are relevant to so parts only relevant to Authorization Agent can be safely skipped.

**Reply by ericprud** [reply]
+1. If we decide to separate, that annotation effort won't have been wasted.

**Reply by justinwb** [reply]
I'm definitely up to looking at how we can break down the spec into easier to digest component parts.

**Reply by elf-pavlik** [reply]
Once we release initial version of sai-js with module for applications I'll mark parts of the spec it conforms to.
